addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/BodyGenerator.java

```diff
@@ -339,4 +346,23 @@ private static String urlEncode(String string) {
             return "";
         }
     }
+
+    private String extractExampleBody(MediaType mediaType) {
+        String exampleBody = null;
+
+        if (mediaType.getExamples() != null
+                && !mediaType.getExamples().isEmpty()
+                && mediaType.getExamples().entrySet().stream().findFirst().get().getValue()
+                        != null) {
+            Example example =
+                    mediaType.getExamples().entrySet().stream().findFirst().get().getValue();
+
+            if (example.getValue() != null) {
+                exampleBody = example.getValue().toString();
+            }
+        } else if (mediaType.getExample() != null) {
```

Comment:
Could you add a test case for this one too?

Reply:
Sure